custom_components/mqtt_vacuum_camera/utils/thread_pool.py

```
@@ -24,87 +26,207 @@
         """
-        executor = self.get_executor(name, max_workers)
-        loop = asyncio.get_event_loop()
-        return await loop.run_in_executor(executor, func, *args)
+        cpu_count = os.cpu_count() or 1
+
+        if task_type == "decompression":
+            # For decompression tasks, use at most 2 workers to avoid excessive resource usage
+            return min(2, cpu_count)
+        else:
+            # For all other tasks, use 1 worker to minimize resource usage
+            return 1
```

Pylint (3.3.7)

[refactor] Unnecessary "else" after "return", remove the "else" and de-indent the code inside it

(R1705)